hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkTableChecker.java

```diff
@@ -148,30 +148,58 @@ public void setTableDataSet(DataSet tableDataSet) {
      * @throws InterruptedException
      * @throws ExecutionException
      */
-    private List<String> checkDuplicateIndexes(PairDataSet index) throws StandardException, InterruptedException, ExecutionException {
-        List<String> messages = Lists.newLinkedList();
-        SpliceSpark.pushScope(String.format("Check duplicates in index %s.%s", schemaName, indexName));
-        JavaPairRDD duplicateIndexRdd = ((SparkPairDataSet)index).rdd
-                .combineByKey(new createCombiner(), new mergeValue(), new mergeCombiners())
-                .filter(new DuplicateIndexFilter());
+    private List<String> checkDuplicateIndexes(PairDataSet table, PairDataSet index) throws StandardException {
+        try {
+            SpliceSpark.pushScope(String.format("Check duplicates in index %s.%s", schemaName, indexName));
+            JavaPairRDD duplicateIndexRdd = ((SparkPairDataSet) index).rdd
+                    .combineByKey(new CreateCombiner(), new MergeValue(), new MergeCombiners())
+                    .filter(new DuplicateIndexFilter());
 
-        long count = 0;
-        JavaFutureAction<List> futureAction = duplicateIndexRdd.takeAsync(maxCheckTableError);
-        List<Tuple2<String, List<byte[]>>> result = futureAction.get();
-        for(Tuple2<String, List<byte[]>> tuple : result) {
-            List<byte[]> keys = tuple._2;
-            for (byte[] key : keys) {
-                indexKeyDecoder.set(key, 0, key.length);
-                indexKeyDecoder.decode(indexKeyTemplate);
-                messages.add(indexKeyTemplate.getClone().toString() + "=>" + tuple._1);
-                count++;
+            JavaPairRDD joinedRdd = duplicateIndexRdd
+                    .join(((SparkPairDataSet) table).rdd);
+
+            JavaRDD duplicateIndex = joinedRdd
+                    .mapPartitions(new SparkFlatMapFunction<>(new DeleteDuplicateIndexFunction<>(conglomerate, txn, tentativeIndex, baseColumnMap, fix)));
+
+            Iterator it = duplicateIndex.toLocalIterator();
+
+            if (fix) {
+                return fixDuplicateIndexes(it);
+            } else {
+                return reportDuplicateIndexes(it);
             }
+        }catch (Exception e) {
+            throw StandardException.plainWrapException(e);
         }
-        if (count >= maxCheckTableError) {
-            count = duplicateIndexRdd.count();
-            messages.add("...");
+    }
```

Comment:
re-add SpliceSpark.popScope()

(this function does SpliceSpark.pushScope , but SpliceSpark.popScope(); was moved (unintentionally i guess) to line 204))